From: Debian Qt/KDE Maintainers <debian-qt-kde@lists.debian.org>
Date: Thu, 11 Oct 2018 14:24:17 +0300
Subject: place_global_config_in_etc

We use DEB_CONFIG_INSTALL_DIR ?= /usr/share/kde4/config but we want global
configuration files in /etc/kde4. This patch hardcodes this value in
config-workspace.cmake and renames the define to KDE_DEBIAN_CONFDIR so that
we can detect possible new uses of KDE_CONFDIR in upstream code that might
have trouble if we apply this patch. With the renamed define, it will at
least FTBFS, so that we have the chance to inspect.
---
 config-workspace.h.cmake        | 2 +-
 kcontrol/dateandtime/helper.cpp | 2 +-
 kdm/config.def                  | 2 +-
 kdm/kcm/helper.cpp              | 6 +++---
 kdm/kcm/main.cpp                | 4 ++--
 kdm/kfrontend/genkdmconf.c      | 5 +++--
 kdm/kfrontend/kdmctl.c          | 2 +-
 ksmserver/shutdowndlg.cpp       | 2 +-
 8 files changed, 13 insertions(+), 12 deletions(-)

diff --git a/config-workspace.h.cmake b/config-workspace.h.cmake
index 90959bf..0d8c810 100644
--- a/config-workspace.h.cmake
+++ b/config-workspace.h.cmake
@@ -129,7 +129,7 @@
 #define KDE_BINDIR "${BIN_INSTALL_DIR}"
 
 /* KDE's configuration directory */
-#define KDE_CONFDIR "${CONFIG_INSTALL_DIR}"
+#define KDE_DEBIAN_CONFDIR "/etc/kde4"
 
 /* KDE's static data directory */
 #define KDE_DATADIR "${DATA_INSTALL_DIR}"
diff --git a/kcontrol/dateandtime/helper.cpp b/kcontrol/dateandtime/helper.cpp
index 9168db3..5f6f35a 100644
--- a/kcontrol/dateandtime/helper.cpp
+++ b/kcontrol/dateandtime/helper.cpp
@@ -58,7 +58,7 @@ int ClockHelper::ntp( const QStringList& ntpServers, bool ntpEnabled,
   int ret = 0;
 
   // write to the system config file
-  QFile config_file(KDE_CONFDIR "/kcmclockrc");
+  QFile config_file(KDE_DEBIAN_CONFDIR "/kcmclockrc");
   if(!config_file.exists()) {
     config_file.open(QIODevice::WriteOnly);
     config_file.close();
diff --git a/kdm/config.def b/kdm/config.def
index 3f038dc..b6584f9 100644
--- a/kdm/config.def
+++ b/kdm/config.def
@@ -29,7 +29,7 @@
 #define RCVERMAJOR 2
 #define RCVERMINOR 4
 
-#define KDMCONF KDE_CONFDIR "/kdm"
+#define KDMCONF KDE_DEBIAN_CONFDIR "/kdm"
 #define KDMDATA KDE_DATADIR "/kdm"
 
 #include <sys/param.h> /* for #define BSD */
diff --git a/kdm/kcm/helper.cpp b/kdm/kcm/helper.cpp
index 2234bca..7637367 100644
--- a/kdm/kcm/helper.cpp
+++ b/kdm/kcm/helper.cpp
@@ -131,9 +131,9 @@ ActionReply Helper::save(const QVariantMap &args)
     QString tempConfigName = args.value("tempkdmrcfile").toString();
     QString tempBackgroundConfigName = args.value("tempbackgroundrcfile").toString();
 
-    QString systemConfigName = QString::fromLatin1(KDE_CONFDIR "/kdm/kdmrc");
+    QString systemConfigName = QString::fromLatin1(KDE_DEBIAN_CONFDIR "/kdm/kdmrc");
     QString systemBackgroundConfigName = KConfig(systemConfigName, KConfig::SimpleConfig)
-            .group("X-*-Greeter").readEntry("BackgroundCfg", KDE_CONFDIR "/kdm/backgroundrc");
+            .group("X-*-Greeter").readEntry("BackgroundCfg", KDE_DEBIAN_CONFDIR "/kdm/backgroundrc");
 
     int code = 0;
 
@@ -167,7 +167,7 @@ ActionReply Helper::managefaces(const QVariantMap &args)
 {
     int subaction = args.value("subaction").toInt();
     QString facesDir =
-        KConfig(QString::fromLatin1(KDE_CONFDIR "/kdm/kdmrc"), KConfig::SimpleConfig)
+        KConfig(QString::fromLatin1(KDE_DEBIAN_CONFDIR "/kdm/kdmrc"), KConfig::SimpleConfig)
             .group("X-*-Greeter").readEntry("FaceDir",
                 QString(KStandardDirs::installPath("data") + "kdm/faces" + '/'));
 
diff --git a/kdm/kcm/main.cpp b/kdm/kcm/main.cpp
index a2a3aaa..191e6ba 100644
--- a/kdm/kcm/main.cpp
+++ b/kdm/kcm/main.cpp
@@ -282,7 +282,7 @@ KConfig *KDModule::createTempConfig()
     QString tempConfigName = pTempConfigFile->fileName();
 
     KConfig *pSystemKDMConfig = new KConfig(
-        QString::fromLatin1(KDE_CONFDIR "/kdm/kdmrc"), KConfig::SimpleConfig);
+        QString::fromLatin1(KDE_DEBIAN_CONFDIR "/kdm/kdmrc"), KConfig::SimpleConfig);
 
     KConfig *pTempConfig = pSystemKDMConfig->copyTo(tempConfigName);
     pTempConfig->sync();
@@ -301,7 +301,7 @@ KSharedConfigPtr KDModule::createBackgroundTempConfig()
 
     QString systemBackgroundConfigName =
         config->group("X-*-Greeter").readEntry(
-            "BackgroundCfg", KDE_CONFDIR "/kdm/backgroundrc");
+            "BackgroundCfg", KDE_DEBIAN_CONFDIR "/kdm/backgroundrc");
 
     KConfig systemBackgroundConfig(systemBackgroundConfigName, KConfig::SimpleConfig);
     KSharedConfigPtr pTempConfig = KSharedConfig::openConfig(tempBackgroundConfigName);
diff --git a/kdm/kfrontend/genkdmconf.c b/kdm/kfrontend/genkdmconf.c
index d10d8c1..a1ebb35 100644
--- a/kdm/kfrontend/genkdmconf.c
+++ b/kdm/kfrontend/genkdmconf.c
@@ -2926,7 +2926,7 @@ fprintfLineWrap(FILE *f, const char *msg, ...)
 
 
 static const char * const oldkdes[] = {
-    KDE_CONFDIR,
+    KDE_DEBIAN_CONFDIR,
     "/opt/kde4/share/config",
     "/usr/local/kde4/share/config",
 
@@ -2934,6 +2934,7 @@ static const char * const oldkdes[] = {
     "/usr/local/kde/share/config",
     "/usr/local/share/config",
     "/usr/share/config",
+    "/etc/kde3",
 
     "/opt/kde3/share/config",
     "/usr/local/kde3/share/config",
@@ -2977,7 +2978,7 @@ int main(int argc, char **argv)
 "    Note that you possibly need to use --no-old-kde to make this take effect.\n"
 "  --old-kde /path/to/old/kde-config-dir\n"
 "    Where to look for the kdmrc of a previously installed kdm.\n"
-"    Default is to scan " KDE_CONFDIR " and\n"
+"    Default is to scan " KDE_DEBIAN_CONFDIR " and\n"
 "    {/usr,/usr/local,{/opt,/usr/local}/{kde4,kde,kde3}}/share/config.\n"
 "  --no-old\n"
 "    Do not look at older xdm/kdm configurations, just create default config.\n"
diff --git a/kdm/kfrontend/kdmctl.c b/kdm/kfrontend/kdmctl.c
index fe3f326..01c7d3c 100644
--- a/kdm/kfrontend/kdmctl.c
+++ b/kdm/kfrontend/kdmctl.c
@@ -148,7 +148,7 @@ main(int argc, char **argv)
 {
     char *dpy = getenv("DISPLAY");
     const char *ctl = getenv("DM_CONTROL");
-    const char *cfg = KDE_CONFDIR "/kdm/kdmrc";
+    const char *cfg = KDE_DEBIAN_CONFDIR "/kdm/kdmrc";
     char *ptr;
     int fd;
 
diff --git a/ksmserver/shutdowndlg.cpp b/ksmserver/shutdowndlg.cpp
index 23ca5b4..263b846 100644
--- a/ksmserver/shutdowndlg.cpp
+++ b/ksmserver/shutdowndlg.cpp
@@ -203,7 +203,7 @@ KSMShutdownDlg::KSMShutdownDlg( QWidget* parent,
     mapSpdMethods->insert("HibernateState", QVariant::fromValue(spdMethods.contains(Solid::PowerManagement::HibernateState)));
     context->setContextProperty("spdMethods", mapSpdMethods);
 
-    QString bootManager = KConfig(KDE_CONFDIR "/kdm/kdmrc", KConfig::SimpleConfig).group("Shutdown").readEntry("BootManager", "None");
+    QString bootManager = KConfig(KDE_DEBIAN_CONFDIR "/kdm/kdmrc", KConfig::SimpleConfig).group("Shutdown").readEntry("BootManager", "None");
     context->setContextProperty("bootManager", bootManager);
 
     QStringList options;
